packages/app-bridge/src/tests/DocumentNavigationTreeDummy.ts

```diff
@@ -0,0 +1,203 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import {
+    type DocumentNavigationItem,
+    type GuidelineDocumentPageHeading,
+    type GuidelineDocumentPage,
+    type GuidelineDocumentPageLink,
+    type GuidelinePageCategory,
+} from '../types';
+
+class DocumentPageCategoryDummy implements GuidelinePageCategory {
+    static with({
+        id,
+        title,
+        slug,
+        children,
+    }: {
+        id?: number;
+        title?: string;
+        slug?: string;
+        children?: (GuidelineDocumentPage | GuidelineDocumentPageLink)[];
+    }): GuidelinePageCategory {
+        return new DocumentPageCategoryDummy(id ?? 1234, title ?? 'Dummy Title', slug ?? 'dummy-slug', children ?? []);
+    }
+
+    type: 'page-category';
+
+    private constructor(
+        private readonly _id: number,
+        private readonly _title: string,
+        private readonly _slug: string,
+        private readonly _children: (GuidelineDocumentPage | GuidelineDocumentPageLink)[],
+    ) {
+        this.type = 'page-category';
+    }
```

Comment:
Let's define these outside the constructor:

```suggestion
    readonly type = 'page-category' as const;
    readonly #id: number;
    readonly #title: string;
    readonly #slug: string;
    readonly #children: (GuidelineDocumentPage | GuidelineDocumentPageLink)[];
    
   constructor(id: number, title: string, slug: string, children: (GuidelineDocumentPage | GuidelineDocumentPageLink)[],
    ) {
        this.#id = id;
        this.#title = title;
        this.#slug = slug;
        this.#children = children;
    }
```

same thing for the others